packages/docusaurus/src/client/normalizeLocation.ts

```diff
@@ -18,8 +20,16 @@ export default function normalizeLocation<T extends Location>(location: T): T {
     };
   }
 
+  // If the location was registered with an `.html` extension, we don't strip it
+  // away, or it will render to a 404 page.
+  const matchedRoutes = matchRoutes(routes, location.pathname);
```

Comment:
also handle `matchRoutes(routes, location.pathname + ".html");` ?

Reply:
I wouldn't do that for now, because it makes less sense.

`/docs/installation.html` is a physical file that exists and can be served, so using this as a location makes sense. If the route is `/docs/installation`, we should try to normalize `/docs/installation.html` to that.

However, if both the route and the physical file is `/docs/installation.html`, then `/docs/installation` doesn't really make sense—it's nowhere to be found. Moreover, I'm worried that this would lead to more edge cases, especially around trailing slash and `index.html`.